### DISCUSSION TITLE
Travamentos constantes

### DISCUSSION THREAD

**Discussion by bernasdev** [dgadelha/Portugol-Webstudio#181] [Ideias e sugestões]
Minha turma do curso de DEV usa o Portugol constantemente para desenvolver a lógica de programação, mas toda a sala está enfrentando o problema do site travar constantemente, seja ao escrever o código ou ao executá-lo. Isso não é problema da internet ou dos computadores, pois já foi testado em outro computador melhor, com uma internet particular e o travamento aconteceu com menos linhas ainda, cerca de 30, enquanto no curso começava a travar com 50 a 70 linhas. Não sou especialista, mas acredito que o WebStudio está mal otimizado. Este é o único problema que estamos enfrentando, gosto muito do Portugol!! Se puder resolver isso, vou gostar mais ainda.
Outra coisa, no aplicativo não aceita variáveis de cadeia no escolha caso, mas no site está aceitando, se puder corrigir isso seria top. tbm

**Comment by dgadelha** [comment]
Consegue me passar um código de Portugol que cause o travamento?

**Reply by bernasdev** [reply]
claro!
[atividade cinema 1.por.txt](https://github.com/dgadelha/Portugol-Webstudio/files/14515325/atividade.cinema.1.por.txt)

**Reply by dgadelha** [reply]
Obrigado! Consegui reproduzir o problema aqui e trago um retorno em breve

**Reply by dgadelha** [reply]
@biribinha0, ainda não consegui identificar a possível causa do problema (algumas ideias em mente, mas preciso fazer mais testes)

Enquanto isso, acabei de lançar uma atualização que faz com que a interface não seja mais bloqueada enquanto o código é checado ou transpilado. A partir de agora, também, os tempos de parse, transpilação e checagem são exibidos na saída do programa.

**Reply by dgadelha** [reply]
Resolvido :)

**Reply by bernasdev** [reply]
Beleza, obrigado!!
